core/src/smartcontracts/isi/asset.rs

```diff
@@ -90,6 +95,8 @@ pub mod isi {
                 *quantity = quantity
                     .checked_add(self.object)
                     .ok_or(MathError::OverflowError)?;
+                #[allow(clippy::cast_precision_loss)]
+                wsv.metrics.tx_amounts.observe(*quantity as f64);
```

Comment:
Why? Truncation would never happen. The only thing that could is precision loss, which we don't care about, since we record a histogram value. It doesn't nearly have the level of precision that would be lost in a 52 bit mantissa to begin with. I'd argue that even `f16` would be overkill precision for anything that a human would be able to discern on a histogram.